service/src/main/java/uk/nhs/adaptors/gpc/consumer/gpc/validation/GpcConfigurationValidator.java

```diff
@@ -0,0 +1,113 @@
+
+        validateSspUrl(config);
+
+        var missingSslProperties = checkForMissingSslProperties(config);
+        if (missingSslProperties.size() == NUMBER_OF_SSL_PROPERTIES) {
+            config.setSslEnabled(false);
+            return true;
+        }
+        if (!missingSslProperties.isEmpty()) {
+            setConstraintViolation(context, MISSING_SSL_PROPERTIES_VIOLATION_MESSAGE, missingSslProperties);
+            return false;
+        }
+
+        var invalidSslProperties = checkSslPropertiesAreValidPemFormat(config);
+        if (!invalidSslProperties.isEmpty()) {
+            setConstraintViolation(context, INVALID_FORMAT_VIOLATION_MESSAGE, invalidSslProperties);
+            return false;
+        }
+
+        config.setSslEnabled(true);
+        return true;
+    }
+
+    private static void setConstraintViolation(ConstraintValidatorContext context, String message, List<String> violations) {
+        String violationMessage = String.format(message, String.join(", ", violations));
+
+        LOGGER.error(violationMessage);
+        context.disableDefaultConstraintViolation();
+        context.buildConstraintViolationWithTemplate(violationMessage).addConstraintViolation();
+    }
+
+    private List<String> checkForMissingSslProperties(GpcConfiguration config) {
+        var missingSslProperty = new ArrayList<String>();
+
+        if (StringUtils.isBlank(config.getClientCert())) {
+            missingSslProperty.add("GPC_CONSUMER_SPINE_CLIENT_CERT");
+        }
+        if (StringUtils.isBlank(config.getClientKey())) {
+            missingSslProperty.add("GPC_CONSUMER_SPINE_CLIENT_KEY");
+        }
+        if (StringUtils.isBlank(config.getRootCA())) {
+            missingSslProperty.add("GPC_CONSUMER_SPINE_ROOT_CA_CERT");
+        }
+        if (StringUtils.isBlank(config.getSubCA())) {
+            missingSslProperty.add("GPC_CONSUMER_SPINE_SUB_CA_CERT");
+        }
+
+        return missingSslProperty;
+    }
+
+    private List<String> checkSslPropertiesAreValidPemFormat(GpcConfiguration config) {
+        var invalidSslProperties = new ArrayList<String>();
+
+        if (isInvalidPemFormat(config.getClientCert())) {
+            invalidSslProperties.add("GPC_CONSUMER_SPINE_CLIENT_CERT");
+        }
+        if (isInvalidPemFormat(config.getClientKey())) {
+            invalidSslProperties.add("GPC_CONSUMER_SPINE_CLIENT_KEY");
+        }
+        if (isInvalidPemFormat(config.getRootCA())) {
+            invalidSslProperties.add("GPC_CONSUMER_SPINE_ROOT_CA_CERT");
+        }
+        if (isInvalidPemFormat(config.getSubCA())) {
+            invalidSslProperties.add("GPC_CONSUMER_SPINE_SUB_CA_CERT");
+        }
+
+        return invalidSslProperties;
+    }
+
+    private boolean isInvalidPemFormat(String sslProperty) {
+        try {
+            PemFormatter.format(sslProperty);
+            return false;
+        } catch (Exception e) {
+            return true;
```

Comment:
I think this exception can be logged so that we can investigate what is wrong with a particular SSL property

Reply:
what about this point @MartinWheelerMT